front/tests/005-operational-studies.spec.ts

```diff
@@ -98,17 +68,46 @@ test.describe('Testing if all mandatory elements simulation configuration are lo
     expect(
       await playwrightRollingstockModalPage.getRollingStockInfoComfort().textContent()
     ).toMatch(/ConfortSStandard/i);
-  });
 
-  test('Select composition code', async () => {
-    await playwrightScenarioPage.openTabByText('Paramètres de simulation');
-    await playwrightScenarioPage.getSpeedLimitSelector.click();
-    await playwrightScenarioPage.getSpeedLimitSelector.locator('input').fill('32');
-    await playwrightScenarioPage.getSpeedLimitSelector
+    // ***************** Test Composition Code *****************
+    await scenarioPage.openTabByText('Paramètres de simulation');
+    await expect(scenarioPage.getSpeedLimitSelector).toBeVisible();
+    await scenarioPage.getSpeedLimitSelector.click();
+    await scenarioPage.getSpeedLimitSelector.locator('input').fill('32');
+    await scenarioPage.getSpeedLimitSelector
       .getByRole('button', { name: 'Voyageurs - Automoteurs - E32C' })
       .click();
-    expect(await playwrightScenarioPage.getSpeedLimitSelector.textContent()).toMatch(
+    expect(await scenarioPage.getSpeedLimitSelector.textContent()).toMatch(
       /Voyageurs - Automoteurs - E32C/i
     );
+
+    // ***************** Test choice Origin/Destination *****************
+    const playwrightMap = new PlaywrightMap(playwrightHomePage.page);
+    await scenarioPage.openTabByText('Itinéraire');
+    await playwrightMap.page.waitForTimeout(2000);
+    const itinerary = scenarioPage.getItineraryModule;
+    await expect(itinerary).toBeVisible();
+    await expect(scenarioPage.getMapModule).toBeVisible();
+
+    // Search and select origin
+    await playwrightMap.openMapSearch();
+    await playwrightMap.searchStation(VARIABLES.originSearch);
+    await playwrightHomePage.page.getByRole('button', { name: VARIABLES.originSearchItem }).click();
+    await playwrightMap.closeMapSearch();
+    await playwrightMap.page.waitForTimeout(100);
+    await playwrightMap.clickOnMap(VARIABLES.originPositionClick);
+    await playwrightMap.clickOnOrigin();
```

Comment:
VERY VERY NICE, thank you !!!! :partying_face: :heart_eyes: 

1 small suggestion: can you create a reusable function to select an origin/destination on the map ? I imagine it like this:
```
interface selectPointOnMapProps {
  playwrightMap: PlaywrightMap;
  playwrightHomePage: PlaywrightHomePage;
  stationName: string;
  stationItemName: string | RegExp;
  positionClick: { x: number; y: numbber };
}

export const selectOrigin = async (props: selectPointOnMapProps) => {
  const selectOrigin = true;
  await selectPointOnMap(props, selectOrigin);
}
export const selectDestination = (props: selectPointOnMapProps) => {
  const selectOrigin = false;
  await selectPointOnMap(props, selectOrigin);
}

const selectPointOnMap = (args: selectPointOnMapProps & { isOrigin: boolean }) => {
  const { playwrightMap, ...} = args;
  await playwrightMap.openMapSearch();
  await playwrightMap.searchStation(stationName);
  await playwrightHomePage.page.getByRole('button', { name: stationItemName }).click();
  await playwrightMap.closeMapSearch();
  await playwrightMap.page.waitForTimeout(100);
  await playwrightMap.clickOnMap(positionClick);
  isOrigin ? await playwrightMap.clickOnOrigin() : await playwrightMap.clickOnDestination();
}
```

Reply:
Very good comment, thank you for your feedback.